apps/fabric-website/src/components/Site/Site.tsx

```diff
@@ -65,39 +65,41 @@ export class Site<TPlatforms extends string = string> extends React.Component<IS
     super(props);
 
     this._async = new Async();
-    this.state = {
-      initialRender: true,
-      activePlatforms: {},
-      platform: 'default' as TPlatforms
-    };
-  }
 
-  public componentDidMount(): void {
+    let activePlatforms: ISiteState<TPlatforms>['activePlatforms'] = {};
+
     const { siteDefinition } = this.props;
     if (siteDefinition.pages) {
       // Get top level pages with platforms.
       const topLevelPages = siteDefinition.pages.filter(page => !!page.platforms).map(page => page.title);
 
       // Get local storage platforms for top level pages.
-      let activePlatforms: ISiteState<TPlatforms>['activePlatforms'];
       try {
         // Accessing localStorage can throw for various reasons
         activePlatforms = JSON.parse(localStorage.getItem('activePlatforms') || '') || {};
       } catch (ex) {
-        activePlatforms = {};
+        // ignore
       }
 
       // Set active platform for each top level page to local storage platform or the first platform defined for that page.
       topLevelPages.forEach(item => {
         activePlatforms[item] = activePlatforms[item] || getPageFirstPlatform(item, siteDefinition);
       });
-
-      // Set the initial state with navigation data and the activePlatforms.
-      this.setState({ ...(this._getNavData() as ISiteState<TPlatforms>), activePlatforms }, this._setActivePlatforms);
-      // Handle hash routing
-      window.addEventListener('hashchange', this._handleRouteChange);
-      this._handleRouteChange();
     }
+
+    this.state = {
+      initialRender: true,
```

Comment:
@ecraig12345 I moved most the code in `componentDidMount` to the constructor, is `initialRender` still needed?

Reply:
Unless you also know how to move the initial this._getNavData() call into the constructor, you should keep the initialRender part. Otherwise the page will flash from the partial first render to the full second render.

Reply:
That's going to take some work for sure. I'll leave it and keep thinking.